queries/kotlin/highlights.scm

```diff
@@ -41,44 +49,55 @@
 
 
 ;; Property access syntax
-; TODO
-; a = get().value etc
-(assignment (navigation_expression (navigation_suffix) @type ))
-; val a = A.get().value etc
-(property_declaration  (navigation_expression (navigation_suffix) @type ))
-; value in `mapOf(A.value to B.value)` etc.
-(infix_expression  (navigation_expression (navigation_suffix) @type ))
+; `value` in `a = get().value`
+(assignment (navigation_expression (navigation_suffix) @property ))
+; `value` in `val a = A.get().value`
+(property_declaration  (navigation_expression (navigation_suffix) @property ))
+; `value` in `mapOf(A.value to B.value)`
+(infix_expression  (navigation_expression (navigation_suffix) @property ))
 ; java in `Car::class.java`
-(value_argument (navigation_expression (navigation_suffix) @type ))
-
-
-;; Interpolated
-
-(interpolated_identifier) @type
-(interpolated_expression) @type
+(value_argument (navigation_expression (navigation_suffix) @method ))
 
 
 ;; Builtin functions
-((simple_identifier) @function.builtin
- (#vim-match? @function.builtin "(^let$|^listOf$|^mapOf$|^map$)"))
-
+;; TODO add more functions
+((simple_identifier) @function.macro
+ (#vim-match? @function.macro "(^let$|^listOf$|^mapOf$|^map$|^downto$)"))
 
+;; TODO add more functions
 ((simple_identifier) @function.builtin
- (#vim-match? @function.builtin "^to$"))
+ (#vim-match? @function.builtin "(^print$|^println$)"))
 
 
-(modifiers (annotation (user_type (type_identifier)))) @attribute
+((simple_identifier) @keyword.operator
+ (#vim-match? @keyword.operator "^to$"))
 
-(annotation) @attribute
 
+;; Numbers
 (integer_literal) @number
 
+;; Booleans
 (boolean_literal) @boolean
 
+;; Annotations
+(annotation (single_annotation) @annotation)
+
+;; Types
+; TODO more type definitions
+(variable_declaration (user_type (type_identifier) @type))
+(callable_reference (type_identifier) @type)
```

Comment:
can't this be reduced to `(type_identifier) @type`

Reply:
that would include annotations and classnames, eg.: `Throws` in `@Throws` and `Test` in `class Test(...)`

Reply:
could probably override that though